### DISCUSSION TITLE
Making subscriptions endpoint configurable

### DISCUSSION THREAD

**Discussion by simonpweller** [Netflix/dgs-framework#339] [Ideas]
I was wondering if you would be open to making the endpoint for subscriptions configurable. Apollo Server uses /graphql and it would be nice to be able to swap out implementations like for like. As far as I can tell it's just a hardcoded string right now, so should hopefully be relatively straightforward. Would you be open to a pull request for that?

**Comment by paulbakker** [comment]
Making it configurable would be fine, but the way it's implemented it couldn't _also_ run on `/graphql`, because there would be multiple handlers for the same path. At least that would need extra work. It would be good to support, but the scope is a little larger than just making it configurable.

**Reply by simonpweller** [reply]
Makes sense - thank you for the quick response

**Reply by Ancient-Dragon** [reply]
Hi @paulbakker, don't suppose you have any updates on whether the subscription endpoint is now configurable / has been added a feature request yet?

**Comment by dzatorsky** [comment]
Also, graphiql doesn't work with subscriptions in DGS because it expects subscriptions to be present on /graphql path, while they are located on /subscriptions path. And there is no way to change the subscription path nor there is a way to customize graphiql.

The error message from graphiql: "Trying to execute subscription on /graphql. Use /subscriptions instead!"

**Reply by berngp** [reply]
Thanks for the ping. We should make this configurable and probably default to what graphiql expects eventually (since it's a breaking change).
